zerver/tests/test_message_edit.py

```diff
@@ -3805,6 +3805,47 @@ def test_move_message_to_stream_and_topic(self) -> None:
         )
         self.assert_json_success(result)
 
+    def test_move_many_messages_to_stream_and_topic(self) -> None:
+        (user_profile, old_stream, new_stream, msg_id, msg_id_later) = self.prepare_move_topics(
+            "iago", "first origin stream", "first destination stream", "first topic"
+        )
+
+        with queries_captured() as queries:
+            result = self.client_patch(
+                f"/json/messages/{msg_id}",
+                {
+                    "propagate_mode": "change_all",
+                    "send_notification_to_old_thread": "true",
+                    "stream_id": new_stream.id,
+                    "topic": "first topic",
+                },
+            )
+            self.assert_json_success(result)
+
+        # Adding more messages should not increase the number of
+        # queries
+        (user_profile, old_stream, new_stream, msg_id, msg_id_later) = self.prepare_move_topics(
+            "iago", "second origin stream", "second destination stream", "second topic"
+        )
+        for i in range(1, 101):
+            self.send_stream_message(
```

Comment:
Sending 100 messages should make this test take a few seconds to run, I think; we try to avoid that in the test suite since slow tests is a silent time waster.